lib/array_intersection.rb

```diff
@@ -1,4 +1,25 @@
-def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+#Design and implement a method that takes two integer arrays with unique values and returns their intersection in a new array.
+ def intersection(list1, list2)
+  return [] if list1.length == 0 || list2.length == 0
 
-end
\ No newline at end of file
+  hash_function = {}
+  result = []
+
+  list1.each do |int|
+   hash_function[int] = false
```

Comment:
I would not use false to indicate something exists, rather I would use true or some other value.